exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -82,21 +79,30 @@ func (r *ClusterResourceSetReconciler) SetupWithManager(ctx context.Context, mgr
 			handler.EnqueueRequestsFromMapFunc(r.resourceToClusterResourceSet),
 			builder.OnlyMetadata,
 			builder.WithPredicates(
-				resourcepredicates.ResourceCreate(ctrl.LoggerFrom(ctx)),
+				predicate.Funcs{
+					CreateFunc:  func(e event.CreateEvent) bool { return true },
+					UpdateFunc:  func(e event.UpdateEvent) bool { return true },
+					DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+					GenericFunc: func(e event.GenericEvent) bool { return false },
+				},
```

Comment:
```suggestion
				resourcepredicates.ResourceCreateOrUpdate(ctrl.LoggerFrom(ctx)),
```
nit: what about `resourcepredicates.ResourceCreateOrUpdate`

Reply:
(looks like after this PR `ResourceCreate` is not used anymore, so we could just rename it)